hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/ServerCall.java

```diff
@@ -175,23 +178,17 @@ private void release(int mask) {
     }
   }
 
-  @Override
-  public void cleanup() {
-    release(0b01);
+  public void retainByWAL() {
+    reference.incrementAndGet();
   }
 
-  public void retainByWAL() {
-    for (;;) {
-      int ref = reference.get();
-      int nextRef = ref | 0b10;
-      if (reference.compareAndSet(ref, nextRef)) {
-        return;
+  public void releaseByWAL() {
+    if (reference.decrementAndGet() == 0) {
```

Comment:
I suggest we add some comments here, to say that decrementAndGet still work even if the highest bit is 1, which means we are decrementing a negative value. I was confused at first but then I realized that the arithmetic is the same for positive and negative value.

Reply:
I did some test to show the binary bits are set correctly.

Reply:
Thx for the suggestion; added comments. Plz check